cpp/src/datetime/datetime_ops.cu

```diff
@@ -643,20 +630,48 @@ std::unique_ptr<column> extract_second(column_view const& column,
 
 std::unique_ptr<cudf::column> extract_datetime_component(cudf::column_view const& column,
                                                          datetime_component component,
+                                                         rmm::cuda_stream_view stream,
                                                          rmm::device_async_resource_ref mr)
 {
   CUDF_FUNC_RANGE();
   switch (component) {
-    case datetime_component::YEAR: return extract_year(column, mr);
-    case datetime_component::MONTH: return extract_month(column, mr);
-    case datetime_component::DAY: return extract_day(column, mr);
-    case datetime_component::WEEKDAY: return extract_weekday(column, mr);
-    case datetime_component::HOUR: return extract_hour(column, mr);
-    case datetime_component::MINUTE: return extract_minute(column, mr);
-    case datetime_component::SECOND: return extract_second(column, mr);
-    case datetime_component::MILLISECOND: return extract_millisecond_fraction(column, mr);
-    case datetime_component::MICROSECOND: return extract_microsecond_fraction(column, mr);
-    case datetime_component::NANOSECOND: return extract_nanosecond_fraction(column, mr);
+    case datetime_component::YEAR:
```

Comment:
Hmm, while I am not usually in favour of classic macros, perhaps here, judiciously applied, it would improve things:

```
#define extract(field) case field: return detail::apply_datatime_op<detail::extract_component_operator<field>, cudf::type_id::INT16>(column, stream, mr)

switch (component) {
    extract(datetime_component::YEAR);
    extract(datetime_componet::MONTH);
    ...;
}
```

Perhaps that not really worth it though.